ArrayToStoreAdapter.js

```diff
@@ -0,0 +1,382 @@
+				return function (item) {return item[prop] > value; };
+			case "gte":
+				return function (item) {return item[prop] >= value; };
+			case "in":
+				return function (item) {return value.indexOf(item[prop]) !== -1; };
+			case "match":
+				return function (item) {return value.test(item[prop]); };
+			case "contains":
+				return function (item) {return this._arrayContains(item[prop], value); };
+			/**
+			 * In the case of "and" and "or" the prop in in fact the first member of the two queries and value the
+			 * second member.
+			 */
+			case "and":
+				var f1 = this._compileFilterQuery(prop);
+				var f2 = this._compileFilterQuery(value);
+				return function (item) {return f1(item) && f2(item); };
+			case "or":
+				f1 = this._compileFilterQuery(prop);
+				f2 = this._compileFilterQuery(value);
+				return function (item) {return f1(item) || f2(item); };
+			default:
+				throw new Error("Unknown filter operation '" + query.type + "'");
+			}
+		},
+		/* jshint maxcomplexity: 10*/
+
+		/**
+		 * Function to test if an array contains all the values in parameter values.
+		 * @param array
+		 * @param values
+		 * @returns {*}
+		 * @private
+		 */
+		_arrayContains: function (array, values) {
+			for (var j = 0; j < values.length; j++) {
+				if (array.indexOf(values[j]) === -1) {
+					return false;
+				}
+			}
+			return true;
+		},
+
+		/**
+		 * Synchronously deliver change records to all listeners registered via `observe()`.
+		 */
+		deliver:  function () {
+			if (this._arrayHandle) {
+				this._arrayHandle.deliver();
+			}
+			if (this._itemHandles.length !== 0) {
+				this._observeCallbackItems && Observable.deliverChangeRecords(this._observeCallbackItems);
+			}
+		},
+
+		/**
+		 * Discard change records for all listeners registered via `observe()`.
+		 */
+		discardChanges: function () {
+			if (this._arrayHandle && this._itemHandles) {
+				this._beingDiscarded = true;
+				this._arrayHandle.deliver();
+				this._observeCallbackItems && Observable.deliverChangeRecords(this._observeCallbackItems);
+				this._beingDiscarded = false;
+				return this;
+			}
+		},
+
+		/**
+		 * Function to remove the observability on the array and its items.
+		 * @private
+		 */
+		untrack: function () {
+			if (this._arrayHandle) {
+				this._arrayHandle.remove();
+			}
+			if (this._itemHandles) {
+				for (var i = 0; i < this._itemHandles.length; i++) {
+					this._itemHandles[i].remove();
+				}
+			}
+		},
+
+
+		/////////////////////////////////////////////////////////////////////////
+		// Functions dedicated to reproduce the behaviour of dstore functions
+		/////////////////////////////////////////////////////////////////////////
+
+		/**
+		 * Called to perform the fetch operation on the collection.
+		 */
+		fetch: function () {
+			return Promise.resolve(this.data);
+		},
+
+		/**
+		 * Called to perform the fetchRange operation on the collection.
+		 * @param {args} - contains the start index and the end index of the fetch.
```

Comment:
Looks like a typo, it should say something like `@param {Object} args -` (or something like that?